### PR TITLE
format time nanoseconds

### PR DESCRIPTION
**Pull request by mbuerki**
formats nanoseconds timestamps for better readability.

### CODE REVIEW

**Comment by mfehr** [comment]
If I read this correctly, this will turn `123456789123` into `123.456789123`? Isn't this a bit misleading, wouldn't this make more sense: `123'456'789'123` or `123'456789123`?

**Comment by mbuerki** [comment]
@mfehr I see your point. I'll change it to `123'123123123`.

**Comment by schneith** [comment]
What do you think of "XXX s YYY ns"? This might make it more self-explanator.

File: aslam_cv_common/test/test-time.cc [hunk 1]
```diff
@@ -51,4 +51,11 @@ TEST(TestTime, TestGetCurrentTime) {
   EXPECT_TRUE(aslam::time::isValidTime(timestamp_now));
 }
 
+TEST(TestTime, TestFormatTimeNanoseconds) {
+  constexpr int64_t time_ns_1 = 1234567890;
+  EXPECT_EQ(aslam::time::formatTimeNanoseconds(time_ns_1), "1s234567890ns");
```
Comment (dymczykm) [inline aslam_cv_common/test/test-time.cc]: can we have space between each unit?

File: aslam_cv_common/include/aslam/common/time.h [hunk 2]
```diff
@@ -72,6 +72,19 @@ inline bool isValidTime(int64_t time) {
   return time != getInvalidTime();
 }
 
+inline std::string formatTimeNanoseconds(const int64_t timestamp_nanoseconds) {
```
Comment (dymczykm) [inline aslam_cv_common/include/aslam/common/time.h]: why is it called format time? isn't this just taking timeNanosecondsToString?

File: aslam_cv_common/include/aslam/common/time.h [hunk 3]
```diff
@@ -72,6 +72,19 @@ inline bool isValidTime(int64_t time) {
...
+  const std::string timestamp_string = std::to_string(timestamp_nanoseconds);
+  const size_t num_digits = timestamp_string.size();
+  if (num_digits < 10u) {
+    return timestamp_string + "ns";
+  }
+
+  const std::string timestamp_with_dot =
+      timestamp_string.substr(0u, num_digits - 9u) + 's' +
+      timestamp_string.substr(num_digits - 9u) + "ns";
+  return timestamp_with_dot;
```
Comment (dymczykm) [inline aslam_cv_common/include/aslam/common/time.h]: there's no dot anymore

File: aslam_cv_common/include/aslam/common/time.h (hunk 2 above)
Comment (mbuerki) [reply aslam_cv_common/include/aslam/common/time.h]: with `timeNanosecondsToString()` I would expect the same output as `std::to_string()`, but this is not the case here. do you have another name in mind which you prefer?

Comment (dymczykm) [reply aslam_cv_common/include/aslam/common/time.h]: but I think this is what happens within C# and so one, toString produces a nice output with units or whatever is specific for a given type. In this case, that would be units. For me, ``format`` doesn't mean anything.

Comment (dymczykm) [reply aslam_cv_common/include/aslam/common/time.h]: I prefer ``timeNanosecondsToString``

**Pull request merged**